pkgs/by-name/ob/obj2voxel/package.nix

```
@@ -0,0 +1,75 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, static ? stdenv.hostPlatform.isStatic
+, config
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "obj2voxel";
+  version = "1.3.4";
+
+  src = (
+    fetchFromGitHub {
+      owner = "Eisenwave";
+      repo = finalAttrs.pname;
+      rev = "refs/tags/v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-FkGQjYm15le1Na+2U1TgK4nNOZNs5bME5J87pjVFpDs=";
+    }
+  ).overrideAttrs (
+    _: {
+      GIT_CONFIG_COUNT = 1;
+      GIT_CONFIG_KEY_0 = "url.https://github.com/.insteadOf";
+      GIT_CONFIG_VALUE_0 = "git@github.com:";
+    }
+  );
+
+  patches = [
+    # pull missing definitions
+    (
+      fetchpatch {
+        name = "add-std-includes.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/3.diff";
+        hash = "sha256-2JeFBx0Ds++BplWXuxOo4KOAW6gXCT/kFI74i0M0ROM=";
+        stripLen = 1;
+        extraPrefix = "voxelio/";
+      }
+    )
+    # fix compile error on aarch64-darwin
+    (
+      fetchpatch {
+        name = "fix-aarch64-darwin-type-mismatch.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/4.diff";
+        hash = "sha256-AuvjpTGBLKN0f47IoxIDkV32g9A/4RbvGBo3ToGiH/w=";
+        stripLen = 1;
+        extraPrefix = "voxelio/";
+      }
+    )
+    # already fixed in upstream git head
+    ./1.3.4-fix-type-mismatch.patch
+  ];
+
+  postPatch = ''
+    echo 'install(TARGETS obj2voxel-cli DESTINATION ''${CMAKE_INSTALL_PREFIX}/bin)' >> CMakeLists.txt
+  '';
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=${if static then "OFF" else "ON"}"
+  ];
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
+    description = "Convert OBJ and STL files to voxels, with support for textures";
+    mainProgram = "obj2voxel";
+    homepage = "https://github.com/Eisenwave/obj2voxel/blob/master/README.adoc";
+    license = licenses.mit;
+    platforms = platforms.all;
```

Maybe
```suggestion
    platforms = platforms.unix;
```
is more appropriate.

---

This seems remarkable, considering that the package does not use any libraries, CMake is known to be very portable, and pure standard C++ even more. So I would have expected that almost any platform can work (modulo maybe patching around some quirks). But since I have no non-unix platform to test here, I'll have to trust your assessment.